docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md

```diff
@@ -383,6 +383,51 @@ database = sql_database().parallelize()
 table = sql_table().parallelize()
 ```
 
+## Column reflection
+
+Columns and their data types are reflected with SQLAlchemy. The SQL types are then mapped to `dlt` types.
+Most types are supported.
+
+The `reflection_level` argument controls how much information is reflected:
+
+- `reflection_level = "minimal"`: Only column names and nullability are detected. Data types are inferred from the data.
+- `reflection_level = "full"`: Column names, nullability, and data types are detected.
+- `reflection_level = "full_with_precision"`: Column names, nullability, data types, and precision/scale are detected.
+
+If the SQL type is unknown or not supported by `dlt` the column is skipped when using the `pyarrow` backend.
+In other backend the type is inferred from data regardless of `reflection_level`, this often works, some types are coerced to strings
+and `dataclass` based values from sqlalchemy are inferred as `complex` (JSON in most destinations).
+
+You can also override the sql type by passing a `type_adapter_callback` function.
+This function takes an `sqlalchemy` data type and returns a new type (or `None` to force the column to be skipped/inferred depending on backend).
+
+This is useful for example when:
+- You're loading a data type which is not supported by the destination (e.g. you need JSON type columns to be coerced to string)
+- You're using an sqlalchemy dialect which uses custom types that don't inherit from standard sqlalchemy types.
+
+Example, when loading timestamps from Snowflake you can make sure they translate to `timestamp` columns in the result schema:
+
+```py
+import dlt
+from snowflake.sqlalchemy import TIMESTAMP_NTZ
+import sqlalchemy as sa
+from sqlalchemy.sql.sqltypes import TypeAdapter
```

Comment:
tests fail with 
`Module "sqlalchemy.sql.sqltypes" has no attribute "TypeAdapter"`

may be you meant `from pydantic import TypeAdapter` ? or what version of sqlalchemy you have?